pkg/klusterlet/clusterinfo/clusterinfo_controller_test.go

```diff
@@ -499,3 +498,174 @@ func newConfigV1Client(version string, failingStatus bool) openshiftclientset.In
+				Type: clusterv1beta1.DistributionTypeOCP,
+				OCP: clusterv1beta1.OCPDistributionInfo{
+					Version:          "v4.6",
+					AvailableUpdates: []string{"v4.6.8", "v4.6.9"},
+					DesiredVersion:   "v4.6",
+					UpgradeFailed:    false,
+					Channel:          "channel 4.6",
+					Desired:          clusterv1beta1.OCPVersionRelease{},
+					VersionAvailableUpdates: []clusterv1beta1.OCPVersionRelease{
+						{
+							Image:   "quay.io/openshift-release-dev/ocp-release@sha256:6ddbf56b7f9776c0498f23a54b65a06b3b846c1012200c5609c4bb716b6bdcdf",
+							URL:     "https://access.redhat.com/errata/RHSA-2020:5259",
+							Version: "4.6.8",
+						},
+						{
+							Image:   "quay.io/openshift-release-dev/ocp-release@sha256:6ddbf56b7f9776c0498f23a54b65a06b3b846c1012200c5609c4bb716b6bdcdf",
+							URL:     "https://access.redhat.com/errata/RHSA-2020:5259",
+							Version: "4.6.9",
+						},
+					},
+					VersionHistory: []clusterv1beta1.OCPVersionUpdateHistory{
+						{
+							Image:    "quay.io/openshift-release-dev/ocp-release@sha256:4d048ae1274d11c49f9b7e70713a072315431598b2ddbb512aee4027c422fe3e",
+							State:    "Completed",
+							Verified: false,
+							Version:  "4.5.11",
+						},
+						{
+							Image:    "quay.io/openshift-release-dev/ocp-release@sha256:4d048ae1274d11c49f9b7e70713a072315431598b2ddbb512aee4027c422fe3e",
+							State:    "Completed",
+							Verified: false,
+							Version:  "4.5.12",
+						},
+					},
+				},
+			},
+			new: &clusterv1beta1.DistributionInfo{
+				Type: clusterv1beta1.DistributionTypeOCP,
+				OCP: clusterv1beta1.OCPDistributionInfo{
+					Version:          "v4.6",
+					AvailableUpdates: []string{"v4.6.9", "v4.6.8"},
+					DesiredVersion:   "v4.6",
+					UpgradeFailed:    false,
+					Channel:          "channel 4.6",
+					Desired:          clusterv1beta1.OCPVersionRelease{},
+					VersionAvailableUpdates: []clusterv1beta1.OCPVersionRelease{
+						{
+							Image:   "quay.io/openshift-release-dev/ocp-release@sha256:6ddbf56b7f9776c0498f23a54b65a06b3b846c1012200c5609c4bb716b6bdcdf",
+							URL:     "https://access.redhat.com/errata/RHSA-2020:5259",
+							Version: "4.6.9",
+						},
+						{
+							Image:   "quay.io/openshift-release-dev/ocp-release@sha256:6ddbf56b7f9776c0498f23a54b65a06b3b846c1012200c5609c4bb716b6bdcdf",
+							URL:     "https://access.redhat.com/errata/RHSA-2020:5259",
+							Version: "4.6.10",
+						},
+					},
+					VersionHistory: []clusterv1beta1.OCPVersionUpdateHistory{
+						{
+							Image:    "quay.io/openshift-release-dev/ocp-release@sha256:4d048ae1274d11c49f9b7e70713a072315431598b2ddbb512aee4027c422fe3e",
+							State:    "Completed",
+							Verified: false,
+							Version:  "4.5.12",
+						},
+						{
+							Image:    "quay.io/openshift-release-dev/ocp-release@sha256:4d048ae1274d11c49f9b7e70713a072315431598b2ddbb512aee4027c422fe3e",
+							State:    "Completed",
+							Verified: false,
+							Version:  "4.5.13",
+						},
+					},
+				},
+			},
+		},
```

Comment:
can you add another two case
1. old is nil
2.  new is nil

Reply:
done